src/devices/audioRecorderClient/audioRecorderClient.cpp

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright (C) 2019 Istituto Italiano di Tecnologia (IIT)
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#define _USE_MATH_DEFINES
+
+#include "audioPlayerWrapper.h"
+#include <yarp/dev/ControlBoardInterfaces.h>
+#include <yarp/os/LogStream.h>
+
+#include <cmath>
+#include <sstream>
+
+using namespace yarp::sig;
+using namespace yarp::dev;
+using namespace yarp::os;
+using namespace std;
+
+
+audioPlayerWrapper::audioPlayerWrapper() : PeriodicThread(DEFAULT_THREAD_PERIOD),
+    m_irender(nullptr),
+    m_period(DEFAULT_THREAD_PERIOD),
+    m_isDeviceOwned(false)
+{}
+
+audioPlayerWrapper::~audioPlayerWrapper()
+{
+    m_irender = nullptr;
+}
+
+/**
+  * Specify which sensor this thread has to read from.
+  */
+
+bool audioPlayerWrapper::attachAll(const PolyDriverList &device2attach)
+{
+    if (device2attach.size() != 1)
+    {
+        yError("audioPlayerWrapper: cannot attach more than one device");
+        return false;
+    }
+
+    yarp::dev::PolyDriver * Idevice2attach = device2attach[0]->poly;
+
+    if (Idevice2attach->isValid())
+    {
+        Idevice2attach->view(m_irender);
+    }
+
+    if (nullptr == m_irender)
+    {
+        yError("audioPlayerWrapper: subdevice passed to attach method is invalid");
+        return false;
+    }
+    attach(m_irender);
+
+    PeriodicThread::setPeriod(m_period);
+    return PeriodicThread::start();
+}
+
+bool audioPlayerWrapper::detachAll()
+{
+    if (PeriodicThread::isRunning())
+    {
+        PeriodicThread::stop();
+    }
+    m_irender = nullptr;
+    return true;
+}
+
+void audioPlayerWrapper::attach(yarp::dev::IAudioRender *irend)
+{
+    m_irender = irend;
+}
+
+void audioPlayerWrapper::detach()
+{
+    if (PeriodicThread::isRunning())
+    {
+        PeriodicThread::stop();
+    }
+    m_irender = nullptr;
+}
+
+bool audioPlayerWrapper::read(yarp::os::ConnectionReader& connection)
+{
+    yarp::os::Bottle in;
+    yarp::os::Bottle out;
+    bool ok = in.read(connection);
+    if (!ok) return false;
+
+    // parse in, prepare out
+    int action = in.get(0).asVocab();
+    int inter  = in.get(1).asVocab();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'inter' is assigned a value that is never used.](https://app.codacy.com/manual/robotology/yarp/pullRequest?prid=5162834)